app/resources/resources.py

```diff
@@ -0,0 +1,446 @@
+from flask import Blueprint, g, jsonify, request
+from flask_restful import Api, Resource
+from app.models.models import BucketLists, BucketListsSchema, db, Items, ItemsSchema, UsersSchema, User
+from sqlalchemy.exc import IntegrityError, SQLAlchemyError
+from marshmallow import ValidationError
+from flask.ext.httpauth import HTTPBasicAuth
+from flask_restful import reqparse
+from functools import wraps
+import pprint
+from flask.ext.restful import inputs
+
+# Define blueprint
+bucket_list = Blueprint('bucketlists', __name__)
+
+# Initialize flask application
+api = Api(bucket_list)
+
+# Define schemas for the models
+bucket_list_schema = BucketListsSchema()
+items_schema = ItemsSchema()
+users_schema = UsersSchema()
+
+# Initialize auth
+auth = HTTPBasicAuth()
+
+
+def handle_exceptions(f):
+    """Decorator to handle Validation and SQLAlchemyErrors"""
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except ValidationError as err:
+            resp = jsonify({"error": err.messages})
+            resp.status_code = 400
+            return resp
+        except IntegrityError as e:
+            resp = jsonify({"error": e.message})
+            resp.status_code = 401
+            return resp
+        except SQLAlchemyError as e:
+            db.session.rollback()
+            resp = jsonify({"error": str(e)})
+            resp.status_code = 400
+            return resp
+    return decorated
+
+
+def handle_register(f):
+    """Decorator to handle Validation error during Registering"""
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except IntegrityError:
+            resp = jsonify({"error": "The username is already taken"})
+            resp.status_code = 401
+            return resp
+    return decorated
+
+
+class BucketList(Resource):
+    """
+    Handles requests to bucketlists.
+
+    Resource urls:
+        '/bucketlists/'
+        '/bucketlists/<bucket_id>'
+
+    Requests Allowed:
+        'GET', 'POST', 'PUT', 'DELETE'
+    """
+
+    @auth.login_required
+    def get(self, bucket_id=0):
+        """
+        Queries bucketlists.
+
+        Args:
+            self,bucket_id
+
+        Return:
+            All or specific bucketlists belonging to the logged in user.
+        """
+
+        search = False  # Flag to determine whether to search.
+
+        # Get page and limit from request url.
+        try:
+            page = int(request.args.get('page'))
+            limit = int(request.args.get('limit'))
+        except:
+            page = 1
```

Comment:
**Code Quality: Nitpick**

Do you not want to catch these exceptions individually? If I set an invalid value for `page`, limit defaults to 20. Not sure this is optimal